src/cephadm/cephadm

```diff
@@ -108,15 +109,26 @@ def port_in_use(port_num):
     # type (int) -> bool
     """Detect whether a port is in use on the local machine - IPv4 and IPv6"""
 
+    def attempt_bind(address):
+        # type (str) -> None
+        try:
+            s.bind((address, port_num))
+        except (socket.error, OSError) as e:  # py2 and py3
+            if e.errno == errno.EADDRINUSE:
+                raise OSError
+            elif e.errno == errno.EADDRNOTAVAIL:
+                return
+        finally:
+            s.close()
+
     try:
         s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        s.bind(("127.0.0.1", port_num))
-        s.close()
+        attempt_bind('0.0.0.0')
+
         s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
-        s.bind(("::1", port_num))
-        s.close()
+        attempt_bind('::/0')
```

Comment:
You're sure? I'm getting an error when trying to bind 8081:

```
In [6]: s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)

In [7]: s.bind(('::/0', 8081))
---------------------------------------------------------------------------
gaierror                                  Traceback (most recent call last)
<ipython-input-7-1935d55732aa> in <module>()
----> 1 s.bind(('::/0', 8081))

gaierror: [Errno -3] Temporary failure in name resolution
```

This works however:
```
In [8]: s.bind(('::', 8081))

In [9]: 
```

Reply:
thanks for catching!